DDD-7.md

```diff
@@ -0,0 +1,153 @@
+* run potentially time-consuming code (e.g. event transformation or serialization) in the dedicated threads
+* allow possible further speedup by optionally disabling total ordering of the messages
+* be well-prepared for future changes and new features, namely
+  * switching to [virtual threads](https://docs.oracle.com/en/java/javase/21/core/virtual-threads.html)
+  * eventually delegate source tasks to external workers via [gRPC](https://grpc.io/) calls (especially to be able to horizontally scale connector load to multiple pods in Kubernetes cluster)
+  * better integration with Debezium k8s operator and UI
+  * providing Debezium engine as a [Quarkus](https://quarkus.io/) extension
+
+Another high-level goal is to adjust the current Debezium testsuite to use the `DebeziumEngine` interface instead of hard-coded `EmebddedEngine` implementation.
+This should allow for easy switching to any other implementation of `DebeziumEngine` and thus making new implementation to be easily tested with the current testsuite.
+
+## Non-goals
+
+* Change [`DebeziumEngine`](https://github.com/debezium/debezium/blob/v2.4.0.Final/debezium-api/src/main/java/io/debezium/engine/DebeziumEngine.java) interface.
+* Do any other changes in the Debezium [API package](https://github.com/debezium/debezium/tree/v2.4.0.Final/debezium-api).
+* Implement any parallelization inside connectors (e.g. tracking database CDC changes in multiple threads).
+* Remove dependency on Kafka Connect API.
+
+### Preserving Kafka Connect model
+
+As the main goal of the Debezium engine is to be able to execute Debezium outside of Kafka, it may be strange why not to take this opportunity to get rid of Kafka dependencies.
+The reason is simple: it would be a too complex change which would not impact only the Debezium engine.
+E.g. removing [`WorkerConfig`](https://github.com/apache/kafka/blob/3.6.0/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java) would require removing [`OffsetBackingStore`](https://github.com/apache/kafka/blob/3.6.0/connect/runtime/src/main/java/org/apache/kafka/connect/storage/OffsetBackingStore.java),
+which would require removing [`OffsetStorageReader`](https://github.com/apache/kafka/blob/3.6.0/connect/api/src/main/java/org/apache/kafka/connect/storage/OffsetStorageReader.java) etc., etc., resulting in substantial changes in the Debezium core and connectors.
+Therefore, this should be done in a separate task which would deserve a dedicated DDD describing all the changes and proposing replacements for Kafka Connect interfaces and classes.
+
+## Proposed changes
+
+### Thread pools
+
+Implementation of concurrent processing will be based on the Java [Executors](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/Executors.html) framework.
+As the Debezium is currently based on Java 11, new concurrency features introduced by the project Loom, namely virtual threads and structured concurrency, cannot be used now.
```

Comment:
As we intend to move to Java 17 in next year, are there any Java 17 features that would help even more with the implementation and further improvements or it is not relevant for the new engine?